### PR TITLE
feat: Send segment userid to intercom

### PR DESCRIPTION
**Pull request by cyaiox**
This PR sends the segment `anonymousId ` to the intercom interface.

### CODE REVIEW

File: src/components/Intercom/Intercom.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import IntercomWidget from 'decentraland-dapps/dist/components/Intercom'
+import { getAnalytics, getAnonymousId } from 'decentraland-dapps/dist/modules/analytics/utils'
+import { config } from 'config'
+
+const APP_ID = config.get('INTERCOM_APP_ID', '')
+
+export const Intercom: React.FC = () => {
+  const [dclAnonymousUserID, setDclAnonymousUserID] = useState('')
+
+  const analytics = getAnalytics()
```
Comment (LautaroPetaccio) [inline src/components/Intercom/Intercom.tsx:11]: Is analytics always going to be defined?

File: src/components/Intercom/Intercom.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  const analyticsReadyCallback = useCallback(() => {
+    const dclAnonymousUserID = getAnonymousId()
+    if (dclAnonymousUserID) {
+      setDclAnonymousUserID(dclAnonymousUserID)
+    }
+  }, [])
+
+  analytics.ready(analyticsReadyCallback)
```
Comment (LautaroPetaccio) [inline src/components/Intercom/Intercom.tsx:20]: Does it matters if the ready value is set multiple times?
```suggestion
  useEffect(() => analytics.ready(analyticsReadyCallback), [])
```

**Review by LautaroPetaccio — COMMENTED** [review]
Left two comments to discuss, but everything look great.

File: src/components/Intercom/Intercom.tsx (hunk 1 above)
Comment (cyaiox) [reply src/components/Intercom/Intercom.tsx:11]: yep, analytics is always going to be defined

File: src/components/Intercom/Intercom.tsx (hunk 2 above)
Comment (cyaiox) [reply src/components/Intercom/Intercom.tsx:20]: @LautaroPetaccio -- thanks for the review! It does not affect `segment+intercom` functionality but should only be set once when the page is first loaded. It'll update the code

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**